imageproxy.go

```diff
@@ -216,7 +215,11 @@ func (p *Proxy) serveImage(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		msg := fmt.Sprintf("error fetching remote image: %v", err)
 		p.log(msg)
-		http.Error(w, msg, http.StatusInternalServerError)
+		if v, ok := err.(*url.Error); ok && v.Unwrap() == errTooManyRedirects {
+			http.Error(w, msg, http.StatusBadRequest)
```

Comment:
I'm not sure that a 400 error makes sense here... there's nothing wrong with the original request, and there's nothing the client could do differently to fix the issue.  I think just leaving this as a 500 makes sense, since the error is entirely internal to imageproxy itself and its redirection policy.

I looked around and couldn't find any prior art for the status code here.  Typically, this kind of policy is enforced in the client itself, so it never needs to get mapped to a status code.  If you know of any, that'd be interesting to see.

Reply:
I agree that `400` is not the right code and I couldn't find a good one for this specific scenario. We don't really need to check for the `400` status code in the client right now so let's just keep the `500`.

I've pushed the changes.